jerry-core/vm/vm.c

```diff
@@ -98,7 +98,11 @@ vm_op_get_value (ecma_value_t object, /**< base object */
       {
         ecma_object_t *obj_p = ecma_get_object_from_value (object);
 
-        return ecma_builtin_object_object_get_prototype_of (obj_p);
+        ecma_value_t has_property = ecma_op_object_has_property (obj_p, property_name_p);
+        if (ecma_is_value_false (has_property))
+        {
+          return ecma_builtin_object_object_get_prototype_of (obj_p);
+        }
```

Comment:
According to ES2015, this thing should be a getter/setter:
http://www.ecma-international.org/ecma-262/6.0/#sec-object.prototype.__proto__

It shouldn't be in the vm at all. If we do this correctly, the issue should go away, since objects will have no such own property.

Reply:
I checked before what happen if I remove __proto__ handling from vm and I got some issues that I could not solve. But I checked it again right now and it looks like it would solve the problem as you said. I will recheck one more time and make changes to this PR.

Reply:
@rwalczyna As upon this issue has been revealed I stardted to lookup the reason of this error. I've already found it and I've already have a fix for it (also includes the removal of the __proto__ handling in vm_get). I'd submit a PR with the fix and I'd mark you as a co-author if it is suitable for you.

Reply:
Related PR https://github.com/jerryscript-project/jerryscript/pull/3704.

Reply:
@rerobika Sounds good, so I think this PR can be closed.